tests/qchem/test_tapering.py

```diff
@@ -629,3 +630,190 @@ def test_taper_obs(symbols, geometry, charge):
             @ scipy.sparse.coo_matrix(state_tapered).T
         ).toarray()
         assert np.isclose(obs_val, obs_val_tapered)
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "charge", "num_commuting"),
+    [
+        (
+            ["H", "H"],
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.40104295]], requires_grad=True),
+            0,
+            (0, 1),
+        ),
+        (
+            ["He", "H"],
+            np.array(
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]],
+                requires_grad=True,
+            ),
+            1,
+            (2, 1),
+        ),
+        (
+            ["H", "H", "H", "H"],
+            np.array(
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, 2.0], [0.0, 0.0, 3.0]],
+                requires_grad=True,
+            ),
+            0,
+            (4, 10),
+        ),
+    ],
+)
```

Comment:
One suggestion here, instead of doing a full workflow test we could just parameterize over the inputs of the taper_operation function for each case. This way we don't have to compute those values each time. 

If you think developers who see the test might want to know where it came from, then we can add comments beside each parameterization saying something like `# these ops come from the H2 molecule with ____ geometry and ____ charge`

This reduces the complexity of the test and focuses on specifically what you want to test.

Reply:
Great idea! I have updated these with their corresponding tapering data.